scripts/build/download-tool.sh

```diff
@@ -19,22 +19,11 @@
 set -u
 set -e
 
-function check_dep() {
-  if ! command -v $1 2>/dev/null 1>&2; then
-    echo "Dependency missing: $1" 1>&2
-    exit 1
-  fi
-}
+./scripts/build/check-dependency.sh wget
 
-check_dep wget
-
-if command -v sha256sum 2>/dev/null 1>&2; then
-  SHA256SUM=sha256sum
-elif command -v shasum 2>/dev/null 1>&2; then
-  SHA256SUM="shasum -a 256"
-else
-  echo "Dependency missing: sha256sum or shasum" 1>&2
-  exit 1
+SHA256SUM=$(./scripts/build/check-dependency.sh sha256sum shasum)
+if [ "$SHA256SUM" == "shasum" ]; then
+  SHA256SUM="$SHA256SUM -a 256"
 fi
```

Comment:
I tried this, however the shell seems to strip the quotes when passing them to `check-dependency.sh`, therefore the options are ignored (unless I'm doing something wrong):

```diff
diff --git a/scripts/build/check-dependency.sh b/scripts/build/check-dependency.sh
index 6df82ec..a5b3346 100755
--- a/scripts/build/check-dependency.sh
+++ b/scripts/build/check-dependency.sh
@@ -16,9 +16,10 @@
 # limitations under the License.
 ###

+set -x
 set -e
 set +u
-ARGV_DEPENDENCIES=$*
+ARGV_DEPENDENCIES=$@
 set -u

 if [ -z "$ARGV_DEPENDENCIES" ]; then
```

Now check this:

```sh
$ ./scripts/build/check-dependency.sh foo "bower -v"
+ set -e
+ set +u
+ ARGV_DEPENDENCIES='foo bower -v'
+ set -u
+ '[' -z 'foo bower -v' ']'
+ RESULT=
+ for dependency in '$ARGV_DEPENDENCIES'
+ command -v foo
+ for dependency in '$ARGV_DEPENDENCIES'
+ command -v bower
+ RESULT=bower
+ break
+ '[' -z bower ']'
+ '[' 2 -ne 1 ']'
+ echo bower
bower
```

I'll merge this, but happy to have another PR with this improvement if we manage to make it work.